clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -377,10 +383,21 @@ private void logAll() {
     }
 
     /**
-     * Log warnings for any unused configurations
+     * Log infos for any unused configurations
      */
     public void logUnused() {
-        for (String key : unused())
+        Set<String> unusedKeys = unused();
+        //Printing unusedKeys to users should remove unknownKeys
+        unusedKeys.removeAll(unknown());
```

Comment:
Could we move this `remove unknown` keys into `unUsed` method, and let `logUnused` only do the `log` thing?

Reply:
I don't want to change the existing behavior of unused() method, otherwise many testCases in AbstractConfigTest will fail.
Only when printing unusedKeys to the user, tell the user exactly which configurations are known but unused

Reply:
For KafkaClient, unknownKeys also belong to unusedKeys, what do you think?

Reply:
Before this PR, the `unUsed` set contains both `unUsed` and `unKnown` configs, because we didn't separate them.
After this PR, we hope we can separate the `unUsed` and `unKnown`, so we created a new method `unKnown`. Under this situation, I don't think it makes sense to make `unUsed` set contains both `unUsed` and `unKnown`. That will just confuse other developers.

Reply:
ok, I agree with you.